### PR TITLE
Fix DADict objects

### PR DESCRIPTION
**Pull request by p-l-**
Particularly, this fixes `ls(Ether, verbose=1)` (regression test added).

### CODE REVIEW

**Comment by p-l-** [comment]
@gpotter2 could you try for me `ls(Ether() / IP(ttl=(5, 10)) / TCP(dport=[80, 443]))` on a Windows installation?

**Comment by gpotter2** [comment]
image

**Comment by p-l-** [comment]
@gpotter2 thanks! The bug seems only to happen when `_PcapWrapper_pypcap` is used. Is it the case on your setup?

**Comment by gpotter2** [comment]
Yes... Sounds weird
I'm going to try to reproduce the appveyor bug and find a way of fixing it.

**Comment by p-l-** [comment]
@gpotter2 thanks! I could also try to find another regression test that does not trigger this bug but I think it's better to fix it!

**Comment by gpotter2** [comment]
@p-l- Alright, apparently we need to build a more more good-looking fake interface, otherwise scapy will crash on it.

Fixed by
```
diff --git a/scapy/arch/windows/__init__.py b/scapy/arch/windows/__init__.py
index 854d0ff..25fb643 100755
--- a/scapy/arch/windows/__init__.py
+++ b/scapy/arch/windows/__init__.py
@@ -786,16 +786,24 @@ def route_add_loopback(routes=None, ipv6=False, iflist=Non
e):
     else:
         if not conf.route.routes:
             return
+    def get_valid_guid():
+        if scapy.consts.LOOPBACK_INTERFACE:
+            return scapy.consts.LOOPBACK_INTERFACE.guid
+        else:
+            for i in IFACES:
+                if not i.is_invalid():
+                    return i.guid
     data = {}
     data['name'] = scapy.consts.LOOPBACK_NAME
     data['description'] = "Loopback"
     data['win_index'] = -1
-    data['guid'] = "{0XX00000-X000-0X0X-X00X-00XXXX000XXX}"
-    data['invalid'] = True
+    data['guid'] = get_valid_guid()
+    data['invalid'] = False
     data['mac'] = '00:00:00:00:00:00'
+    data['pcap_name'] = six.text_type("\\Device\\NPF_" + data['guid'])
     adapter = NetworkInterface(data)
     if iflist:
-        iflist.append(six.text_type("\\Device\\NPF_" + adapter.guid))
+        iflist.append(adapter.pcap_name)
         return
     # Remove all LOOPBACK_NAME routes
     for route in list(conf.route.routes):
```

**Comment by p-l-** [comment]
@gpotter2 I though I had applied your patch. What do you propose instead?

File: scapy/arch/windows/__init__.py [hunk 1]
```diff
@@ -772,6 +772,16 @@ def get_working_if():
         # no route
         return scapy.consts.LOOPBACK_INTERFACE
 
+conf.iface = get_working_if()
```
Comment (gpotter2) [inline scapy/arch/windows/__init__.py]: Sorry i was unclear.
Every thing is good except `conf.iface = get_working_if()`

Comment (p-l-) [reply scapy/arch/windows/__init__.py]: Do you want `conf.iface` to be set at runtime (even if you don't need it)? Where would you want this call to be made instead?

Comment (gpotter2) [reply scapy/arch/windows/__init__.py]: It's already done:
https://github.com/secdev/scapy/blob/master/scapy/route.py#L195

Comment (p-l-) [reply scapy/arch/windows/__init__.py]: OK so it's safe to remove it?

**Pull request merged**